### ISSUE TITLE
Opaque URLs should be normalized too

### ISSUE THREAD

**Issue by opennota** [PuerkitoBio/purell#8]
purell doesn't normalize "opaque" `URL`s (see [documentation for url.URL](http://golang.org/pkg/net/url/#URL)).

``` go
package main

import (
	"fmt"
	"net/url"

	"github.com/PuerkitoBio/purell"
)

func main() {
	u := &url.URL{Scheme: "http", Opaque: "//eXAMPLe.com/%3f"}
	fmt.Println(purell.NormalizeURL(u, purell.FlagLowercaseHost|purell.FlagUppercaseEscapes))
}
```

Output: `http://eXAMPLe.com/%3f`

**Comment by PuerkitoBio** [comment]
I guess it should, to be honest I'm not sure I completely grasp the opaque field. I don't really have a need for it, but if someone cares enough to provide a PR (with tests), I'll be happy to merge.

**Comment by pkieltyka** [comment]
I feel this is an important issue and here's something related:
```go
u, _ := url.Parse("www.youtube.com")
fmt.Println(purell.NormalizeURL(u, purell.FlagsUsuallySafeGreedy))
```

will print `/www.youtube.com`

**Comment by PuerkitoBio** [comment]
This is indeed an important issue and is addressed in PR #5 (merged in fix-relative-paths branch for now, in master hopefully soon). This is unrelated to the Opaque field that is the subject of this issue, though (see e.g. http://play.golang.org/p/VP8o_CjrPA).

**Comment by PuerkitoBio** [comment]
Closing, I think the Opaque field is meant to be left untouched. The other issue discussed here by pkieltyka is fixed in master now.
